packages/gatsby/src/utils/parcel/compile-gatsby-files.ts

```diff
@@ -57,6 +59,41 @@ export async function compileGatsbyFiles(
   retry: number = 0
 ): Promise<void> {
   try {
+    // Check for gatsby-node.jsx and gatsby-node.tsx (or other misnamed variations)
+    const files = await readdir(siteRoot)
+
+    let nearMatch = ``
+    const configName = `gatsby-node`
+
+    for (const file of files) {
+      if (nearMatch) {
+        break
+      }
+
+      const { name } = path.parse(file)
+      // Of course, allow valid gatsby-node files
+      if (file === `gatsby-node.js` || file === `gatsby-node.ts`) {
+        break
+      }
+
+      if (isNearMatch(name, configName, 3)) {
+        nearMatch = file
+      }
```

Comment:
This will panic if there are both valid and invalid files. (i.e. both `gatsby-node.ts` and `gatsby-node.tsx`). 

If we want to just use valid one and ignore invalid in this case, we should reset `nearMatch` when we find valid file before breaking. 

If we still want to panic, I think maybe error message need to be a little bit different based on wether there is valid file available or not (so message suggest rename if there is no valid, but if there is both valid and invalid then msg probably should say that valid one will be used and user should remove invalid one?

Reply:
(not blocking for PR - just putting a question up - we can ship it as-is)

Reply:
That's (also) our current gatsby-config behavior. If we'd wanna fix it, we should do it in both places